staging/src/k8s.io/client-go/tools/record/fake.go

```diff
@@ -54,7 +55,12 @@ func (f *FakeRecorder) Eventf(object runtime.Object, eventtype, reason, messageF
 }
 
 func (f *FakeRecorder) AnnotatedEventf(object runtime.Object, annotations map[string]string, eventtype, reason, messageFmt string, args ...interface{}) {
-	f.Eventf(object, eventtype, reason, messageFmt, args...)
+	if f.IncludeAnnotations {
```

Comment:
I feel like `AnnotatedEventf` should always include the annotations. Is there perhaps a scenario where you wouldn't want that?

Reply:
No, but I don't want to make a change to the API for existing users, I'm happy to make inclusion of annotations the default behaviour if you'd prefer - just can't guarantee this won't break people's tests :P

Reply:
I would say that it should be fine to change the output here since the fact that the current fake implementation doesn't output any information about annotations is a bug since it is impossible for controllers today to know if their annotations are being set properly when emitting an event.
Also, `AnnotatedEventf` is not frequently used so the numbers of users that would have to adapt to this change should be fairly small.

@aojea do you perhaps know what is the stance towards changes in fake libraries that could break end-users tests?

Reply:
But you would still need to fix the tests which are broken by this change.